File: flutter_zoom_videosdk-2.4.0/android/src/main/java/com/flutterzoom/videosdk/FlutterZoomVideoSdkRemoteCameraControlHelper.java
package com.flutterzoom.videosdk;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkErrors;

import java.util.Map;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import us.zoom.sdk.ZoomVideoSDKErrors;
import us.zoom.sdk.ZoomVideoSDKRemoteCameraControlHelper;
import us.zoom.sdk.ZoomVideoSDKUser;

public class FlutterZoomVideoSdkRemoteCameraControlHelper {

    private Activity activity;

    FlutterZoomVideoSdkRemoteCameraControlHelper(Activity activity) {
        this.activity = activity;
    }

    private ZoomVideoSDKRemoteCameraControlHelper getRemoteCameraControlHelper(String userId) {
        ZoomVideoSDKRemoteCameraControlHelper remoteCameraControlHelper = null;
        try {
            ZoomVideoSDKUser user = FlutterZoomVideoSdkUser.getUser(userId);
            remoteCameraControlHelper = user.getRemoteCameraControlHelper();
            if (remoteCameraControlHelper == null) {
                throw new Exception("No Remote Camera Control Helper Found");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return remoteCameraControlHelper;
    }

    public void requestControlRemoteCamera(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        String userId = (String) params.get("userId");
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getRemoteCameraControlHelper(userId).requestControlRemoteCamera()));
            }
        });
    }

    public void giveUpControlRemoteCamera(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        String userId = (String) params.get("userId");
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getRemoteCameraControlHelper(userId).giveUpControlRemoteCamera()));
            }
        });
    }

    public void turnLeft(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        int range = (Integer) params.get("range");
        String userId = (String) params.get("userId");
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getRemoteCameraControlHelper(userId).turnLeft(range)));
            }
        });
    }

    public void turnRight(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        int range = (Integer) params.get("range");
        String userId = (String) params.get("userId");
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getRemoteCameraControlHelper(userId).turnRight(range)));
            }
        });
    }

    public void turnDown(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        int range = (Integer) params.get("range");
        String userId = (String) params.get("userId");
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getRemoteCameraControlHelper(userId).turnDown(range)));
            }
        });
    }

    public void turnUp(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        int range = (Integer) params.get("range");
        String userId = (String) params.get("userId");
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getRemoteCameraControlHelper(userId).turnUp(range)));
            }
        });
    }

    public void zoomIn(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        int range = (Integer) params.get("range");
        String userId = (String) params.get("userId");
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getRemoteCameraControlHelper(userId).zoomIn(range)));
            }
        });
    }

    public void zoomOut(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        int range = (Integer) params.get("range");
        String userId = (String) params.get("userId");
        if (userId == null || userId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "userId is null or empty", null);
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getRemoteCameraControlHelper(userId).zoomOut(range)));
            }
        });
    }
}

File: flutter_zoom_videosdk-2.4.0/ios/Classes/FlutterZoomVideoSdkRemoteCameraControlHelper.m

#import "FlutterZoomVideoSdkRemoteCameraControlHelper.h"
#import "JSONConvert.h"
#import "FlutterZoomVideoSdkUser.h"


@implementation FlutterZoomVideoSdkRemoteCameraControlHelper

- (ZoomVideoSDKRemoteCameraControlHelper *)getCameraControllerHelper:(NSString *)userId
{
    ZoomVideoSDKRemoteCameraControlHelper* cameraControlHelper = nil;
    @try {
        ZoomVideoSDKUser* user = [FlutterZoomVideoSdkUser getUser:userId];
        cameraControlHelper = [user getRemoteCameraControlHelper];

        if (cameraControlHelper == nil) {
            NSException *e = [NSException exceptionWithName:@"NoCameraControlHelper" reason:@"No Camera Control Helper Found" userInfo:nil];
            @throw e;
        }
    } @catch(NSException *e) {
        NSLog(@"%@ - %@", e.name, e.reason);
    }
    return cameraControlHelper;
}

-(void) requestControlRemoteCamera: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString* userId = call.arguments[@"userId"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getCameraControllerHelper: userId] requestControlRemoteCamera])]);
    });
}

-(void) giveUpControlRemoteCamera: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString* userId = call.arguments[@"userId"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getCameraControllerHelper: userId] giveUpControlRemoteCamera])]);
    });
}

-(void) turnLeft: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString* userId = call.arguments[@"userId"];
        NSNumber* rangeNum = call.arguments[@"range"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        int range = [rangeNum intValue];
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getCameraControllerHelper: userId] turnLeft:range])]);
        
    });
}

-(void) turnRight: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString* userId = call.arguments[@"userId"];
        NSNumber* rangeNum = call.arguments[@"range"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        int range = [rangeNum intValue];
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getCameraControllerHelper: userId] turnRight:range])]);
    });
}

-(void) turnUp: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString* userId = call.arguments[@"userId"];
        NSNumber* rangeNum = call.arguments[@"range"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        int range = [rangeNum intValue];
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getCameraControllerHelper: userId] turnUp:range])]);
    });
}

-(void) turnDown: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString* userId = call.arguments[@"userId"];
        NSNumber* rangeNum = call.arguments[@"range"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        int range = [rangeNum intValue];
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getCameraControllerHelper: userId] turnDown:range])]);
    });
}

-(void) zoomIn: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString* userId = call.arguments[@"userId"];
        NSNumber* rangeNum = call.arguments[@"range"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        int range = [rangeNum intValue];
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getCameraControllerHelper: userId] zoomIn:range])]);
    });
}

-(void) zoomOut: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString* userId = call.arguments[@"userId"];
        NSNumber* rangeNum = call.arguments[@"range"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        int range = [rangeNum intValue];
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getCameraControllerHelper: userId] zoomOut:range])]);
    });
}

@end
